typed-racket-lib/typed-racket/utils/prefab-c.rkt

```diff
@@ -10,51 +10,33 @@
 
 ;; name
 (define (prefab/c-name ctc)
-  (define key (base-prefab/c-key ctc))
-  (define field-contracts (base-prefab/c-field-contracts ctc))
-  `(prefab/c ,key ,@(map contract-name field-contracts)))
+  `(prefab/c ',(prefab-struct-type-key (base-prefab/c-struct-type ctc))
+             ,@(for/list ([spec (in-list (base-prefab/c-field-specs ctc))])
+                 (contract-name (field-spec-contract spec)))))
+
+(define (prefab-struct-type-key prefab-struct-type)
```

Comment:
As far as I can tell, yes. It does seem like it would make sense as a primitive, though. (It seems in general like the standard library is a bit thin on prefab operations.)